SparkPost/2web2ui
# TR-1345 - Send test email

[TR-1345](https://jira.int.messagesystems.com/browse/TR-1345) - Sending a Test Email from Templates V2 UI.

### What Changed
- Adds new `tohaveTextContent()` Jest matcher (full credit to @bkemper)
- Addresses small issue with templates selector that was returning template `text` as `undefined`
- Adds new `className` prop to the `<Loading/>` component
- Adds new `<MultiEmailField/>` re-usable component with:
    * `useMultiEmailField` hook
    * documentation
    * supporting tests
- Exposes the `sendPreview` and `showAlert` action for use on `<EditAndPreviewPage/>`
- Adds new screen reader only content to `<PreviewControlBar/>` buttons
- Adds relevant HTML attributes to `<PreviewControlBar/>` buttons to make them keyboard focusable
- Adds new styles to `<PreviewControlBar/>` to ensure the active state is not communicated through color alone
- Adds new `<SendTestEmalButton/>` component that renders the "Send a Test" button and relevant Modal with form elements to send a preview email

### How To Test
- Navigate to `http://localhost:3100/templatesV2`
- Click on an email template in the list
- Click on 'Send a Test'
- Send a test email using the UI
- Things to check:
    * The `<MultiEmailField/>` component has some fancy interaction! Spacebar and blur adds the text entry to the email list, errors are thrown if entry is not a valid email
    * Submitting the form throws an error if no entry is present
    * Opening the modal saves the state of the draft
    * Sending a preview closes the modal automatically
    * Check the unit tests for additional test cases that can be checked manually

### To Do
- [X] Ensure the draft saves when the user opens the preview modal
